### DISCUSSION TITLE
yolov8 when call predict.  How to get the run/detect/ folder of the run number

### DISCUSSION THREAD

**Discussion by jjwallaby** [ultralytics/ultralytics#7582] [General]
Hi,

WRT new yolov8 when call predict.  How to get the run/detect/ folder of the run number ?

# Run batched inference on a list of images
results = model('image1.jpg', save_crop=True)  # return a list of Results objects

Output:
image 1/1 D:\tensorflow39\yolov8tst\image1.jpg: 320x320 1 driverlicense, 85.2ms
Speed: 2.0ms preprocess, 85.2ms inference, 4.0ms postprocess per image at shape (1, 3, 320, 320)
Results saved to D:\runs\detect\**train42**

How to get the train42 sub-folder name from **results** ?

Thank You

**Comment by pderrenger** [comment]
@jjwallaby hello,

When you run the `predict` method with `save_crop=True`, the results are saved in a new folder within the `runs/detect/` directory. Each run creates a unique sub-folder, usually named with an incrementing run number like `exp`, `exp2`, `exp3`, and so on.

To retrieve the path of the folder where the results are saved, you can access the `results.run_dir` attribute after the prediction. Here's an example:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8n.pt')

# Run batched inference on a list of images
results = model('image1.jpg', save_crop=True)

# Get the path to the saved results
run_directory = results.run_dir
print(f"Results saved to {run_directory}")
```

This will print the directory where the results are saved, which you can then use to access your cropped images or any other saved data. 😊

Keep in mind that the `run_dir` attribute is available after the prediction has been made and the results have been saved.

**Reply by Utsav-2301** [reply]
hi @pderrenger I'm running a basic yolo code in some directory, but the results are getting saved in some other directory's runs folder, which I'm not able to understand why cuz I've freshly cloned the ultralytics repo and then running the code, without any dependencies of that folder.

**Reply by pderrenger** [reply]
Hi @Utsav-2301,

Thank you for reaching out! It sounds like the results directory might not be what you expected. By default, YOLO saves results in a `runs/detect/` directory within your current working directory. If you want to specify a custom directory for saving results, you can use the `project` and `name` arguments in your `predict` call.

Here's an example of how you can specify the directory:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8n.pt')

# Run inference and specify the project and name for the results directory
results = model('image1.jpg', save_crop=True, project='my_custom_project', name='my_run')

# Get the path to the saved results
run_directory = results.run_dir
print(f"Results saved to {run_directory}")
```

This will save your results in `my_custom_project/my_run` instead of the default `runs/detect/`.

If the issue persists, please ensure you are using the latest version of the Ultralytics package. If you continue to encounter problems, providing a minimum reproducible example would be very helpful. You can find more information on how to create one here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Feel free to reach out if you have any more questions! 😊

**Reply by Utsav-2301** [reply]
the issue is results are getting saved in a whole new directory's runs folder, my Ultralytics folder is in D drive and the location where the results are being saved is C drive, which is quite absurd, I tried to find the config file where the path could've been mentioned, but wasn't able to find it.

**Reply by pderrenger** [reply]
Hi @Utsav-2301,

Thank you for bringing this to our attention. It sounds like the results directory is not being set as expected. By default, YOLO saves results in the `runs/detect/` directory within the current working directory. However, if the results are being saved to a different drive, it might be due to an environment variable or a previous configuration.

To explicitly set the directory for saving results, you can use the `project` and `name` arguments in your `predict` call. Here's an example:

```python
from ultralytics import YOLO

# Load your model
model = YOLO('yolov8n.pt')

# Run inference and specify the project and name for the results directory
results = model('image1.jpg', save_crop=True, project='D:/my_custom_project', name='my_run')

# Get the path to the saved results
run_directory = results.run_dir
print(f"Results saved to {run_directory}")
```

This will ensure that your results are saved in the specified directory on the D drive.

If the issue persists, please provide a minimum reproducible example so we can better understand and address the problem. You can find more information on how to create one here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

Additionally, please ensure you are using the latest version of the Ultralytics package.

Feel free to reach out if you have any more questions! 😊
